packages/model-fragments/lib/fragments/ext.js

```diff
@@ -152,83 +100,141 @@ Model.reopen({
+});
+
+/**
+  If the model `hasDirtyAttributes` this function will discard any unsaved
+  changes, recursively doing the same for all fragment properties.
+
+  Example
+
+  ```javascript
+  record.get('name'); // 'Untitled Document'
+  record.set('name', 'Doc 1');
+  record.get('name'); // 'Doc 1'
+  record.rollbackAttributes();
+  record.get('name'); // 'Untitled Document'
+  ```
 
-      this.send('becomeDirty');
+  @method rollbackAttributes
+*/
+decorateMethod(InternalModelPrototype, 'rollbackAttributes', function rollbackFragments() {
+  for (var key in this._fragments) {
+    if (this._fragments[key]) {
+      this._fragments[key].rollbackAttributes();
     }
-  },
+  }
+});
+
+/**
+  If the adapter did not return a hash in response to a commit,
+  merge the changed attributes and relationships into the existing
+  saved data and notify all fragments of the commit.
+
+  @method adapterDidCommit
+*/
+decorateMethod(InternalModelPrototype, 'adapterDidCommit', function adapterDidCommit(returnValue, args) {
+  var attributes = (args[0] && args[0].attributes) || {};
+  var fragment;
+
+  // Notify fragments that the record was committed
+  for (var key in this._fragments) {
+    if (fragment = this._fragments[key]) {
+      fragment._adapterDidCommit(attributes[key]);
+    }
+  }
+});
 
+/**
+  @class JSONSerializer
+  @namespace DS
+*/
+JSONSerializer.reopen({
   /**
-    @method fragmentDidReset
+    Enables fragment properties to have custom transforms based on the fragment
+    type, so that deserialization does not have to happen on the fly
+
+    @method transformFor
     @private
-    */
-  fragmentDidReset: function(key, fragment) {
-    // Make sure there's no entry in the owner record's
-    // `_attributes` hash to indicate the fragment is dirty
-    delete this._attributes[key];
-
-    // Don't reset if the record is new, otherwise it will enter the 'deleted' state
-    // NOTE: This case almost never happens with attributes because their initial value
-    // is always undefined, which is *usually* not what attributes get 'reset' to
-    if (!get(this, 'isNew')) {
-      this.send('propertyWasReset', key);
+  */
+  transformFor: function(attributeType) {
+    if (attributeType.indexOf('-mf-') === 0) {
+      return getFragmentTransform(this.container, this.store, attributeType);
     }
+
+    return this._super.apply(this, arguments);
   }
 });
 
-export { Store, Model };
+// Retrieve or create a transform for the specific fragment type
+function getFragmentTransform(container, store, attributeType) {
+  var registry = container._registry || container;
+  var containerKey = 'transform:' + attributeType;
+  var match = attributeType.match(/^-mf-(fragment|fragment-array|array)(?:\$([^$]+))?(?:\$(.+))?$/);
+  var transformType = match[1];
+  var modelName = match[2];
+  var polymorphicTypeProp = match[3];
+
+  if (!registry.has(containerKey)) {
+    var transformClass = container.lookupFactory('transform:' + transformType);
```

Comment:
Oh, could it be that it's failing because the npm package is still 0.3.3 but the bower is set to this one? Not sure how ember-cli testing uses bower vs npm packages.